modules/ballerina-http/src/main/ballerina/ballerina/net/http/request.bal

```diff
@@ -1,91 +1,292 @@
+    mime:Entity entity = request.getEntity();
+    return mime:getJson(entity);
+}
+
+@Description {value:"Gets the outbound request payload in JSON format"}
+@Param {value:"request: The outbound request message"}
+@Return {value:"The JSON reresentation of the message payload"}
+public function <OutRequest request> getJsonPayload () (json) {
+    mime:Entity entity = request.getEntity();
+    return mime:getJson(entity);
+}
+
+@Description {value:"Gets the inbound request payload in XML format"}
+@Param {value:"request: The inbound request message"}
+@Return {value:"The XML representation of the message payload"}
+public function <InRequest request> getXmlPayload () (xml) {
+    mime:Entity entity = request.getEntity();
+    return mime:getXml(entity);
+}
+
+@Description {value:"Gets the outbound request payload in XML format"}
+@Param {value:"request: The outbound request message"}
+@Return {value:"The XML representation of the message payload"}
+public function <OutRequest request> getXmlPayload () (xml) {
+    mime:Entity entity = request.getEntity();
+    return mime:getXml(entity);
+}
+
+@Description {value:"Gets the inbound request payload as a string"}
+@Param {value:"request: inbound request message"}
+@Return {value:"The string representation of the message payload"}
+public function <InRequest request> getStringPayload () (string) {
+    mime:Entity entity = request.getEntity();
+    return mime:getText(entity);
+}
+
+@Description {value:"Gets the outbound request payload as a string"}
+@Param {value:"request: outbound request message"}
+@Return {value:"The string representation of the message payload"}
+public function <OutRequest request> getStringPayload () (string) {
+    mime:Entity entity = request.getEntity();
+    return mime:getText(entity);
+}
+
+@Description {value:"Gets the inboundrequest payload in blob format"}
+@Param {value:"request: The inbound request message"}
+@Return {value:"The blob representation of the message payload"}
+public function <InRequest request> getBinaryPayload () (blob) {
+    mime:Entity entity = request.getEntity();
+    return mime:getBlob(entity);
+}
+
+@Description {value:"Gets the outbound request payload in blob format"}
+@Param {value:"request: The outbound request message"}
+@Return {value:"The blob representation of the message payload"}
+public function <OutRequest request> getBinaryPayload () (blob) {
+    mime:Entity entity = request.getEntity();
+    return mime:getBlob(entity);
+}
+
+@Description {value:"Gets the form parameters from the HTTP request as a map"}
+@Param {value:"req: The inbound request message"}
+@Return {value:"The map of form params"}
+public function <InRequest request> getFormParams () (map) {
+    mime:Entity entity = request.getEntity();
+    string formData = mime:getText(entity);
+    map parameters = {};
+    if (formData != null && formData != "") {
+        string[] entries = formData.split("&");
+        int i = 0;
+        while (i < lengthof entries) {
```

Comment:
Is it possible to give a proper name for this "i" variable.

Reply:
Ack